src/http/index.js

```diff
@@ -14,9 +16,17 @@ const WS = require('libp2p-websockets')
 const Bootstrap = require('libp2p-bootstrap')
 const errorHandler = require('./error-handler')
 
-function uriToMultiaddr (uri) {
-  const ipPort = uri.split('/')[2].split(':')
-  return `/ip4/${ipPort[0]}/tcp/${ipPort[1]}`
+function hapiInfoToMultiaddr (info) {
+  let hostname = info.host
+  let uri = info.uri
+  // ipv6 fix
+  if (hostname.includes(':') && !hostname.startsWith('[')) {
+    // hapi 16 produces invalid URI for ipv6
```

Comment:
Is this still needed? We use hapi 18 now...

Reply:
Sadly still needed: I tried with vanilla hapi 18 and `uri` inside of `apiServer.info` was still broken (missing brackets), which produced broken `ma`:

```json
{
  "created": 1549909555055,
  "started": 1549909555222,
  "host": "::1",
  "port": 5002,
  "protocol": "http",
  "id": "foo:5989:js0o4qjz",
  "uri": "http://::1:5002",
  "address": "::1",
  "ma": "/ip4//tcp/"
}
```


Due to this the line should stay :)

```suggestion
    // hapi 18 produces invalid URI for ipv6
```